### PR TITLE
Add jupyterlab extension transient-display-data to conda-forge

### PR DESCRIPTION
**Pull request by BoPeng**
Checklist

- [ x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ x] Source is from official source
- [ x] Package does not vend other packages
- [x ] Build number is 0
- [x ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

Mimics https://github.com/conda-forge/jupyterlab-plotly-extension-feedstock , in preparation for adding [jupyterlab-sos](https://github.com/vatlab/jupyterlab-sos) to conda-forge.

### CODE REVIEW

**Comment by BoPeng** [comment]
Got a strange message saying license file not found. Changing `LICENSE` to `LICENSE.txt` did not help. Must have been something trivial... feeling awkward to ask @conda-forge/staged-recipes for help.

**Comment by chrisburr** [comment]
No need to feel awkward :)

There isn't a licence file in the repository: https://github.com/vatlab/transient-display-data

**Comment by BoPeng** [comment]
I see, the license information is in https://github.com/vatlab/transient-display-data/blob/master/package.json but I am adding a LICENSE file now.

**Comment by chrisburr** [comment]
Most common licences, including BSD-3-Clause, require the licence text to be included with all copies of the software. (Which is why conda-forge has this requirement.)

**Comment by BoPeng** [comment]
This one is ready to be reviewed.

File: recipes/jupyterlab-transient-display-data/bld.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
```
Comment (chrisburr) [inline recipes/jupyterlab-transient-display-data/bld.sh]: This file should be called `build.sh`

File: recipes/jupyterlab-transient-display-data/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    - jupyterlab
+    - python >=3.5
+  run:
+    - jupyterlab
+    - nodejs
+    - python >=3.5
+
+test:
+  commands:
+    - jupyter labextension list
```
Comment (chrisburr) [inline recipes/jupyterlab-transient-display-data/meta.yaml]: I don't see how these commands are testing the package? Linux and macOS should be failing as they're empty due to the incorrect build script filename.

File: recipes/jupyterlab-transient-display-data/bld.sh (hunk 1 above)
Comment (BoPeng) [reply recipes/jupyterlab-transient-display-data/bld.sh]: Done.

File: recipes/jupyterlab-transient-display-data/meta.yaml (hunk 2 above)
Comment (BoPeng) [reply recipes/jupyterlab-transient-display-data/meta.yaml]: Added `| grep transient-display-data` to test successful installation of the extension. The problem exists in other jupyterlab extension recipes though, for example, https://github.com/conda-forge/jupyterlab-plotly-extension-feedstock/blob/master/recipe/meta.yaml#L30, https://github.com/conda-forge/jupyterlab-git-feedstock/blob/master/recipe/meta.yaml#L34

**Pull request merged**